plugin/my_asset_manager/MyAssetManagerInterface.py

```diff
@@ -121,6 +119,95 @@ def isEntityReferenceString(self, someString, hostSession):
+        # a write access mode via calling the error callback.
+        if entityTraitsAccess != EntityTraitsAccess.kRead:
+            result = BatchElementError(
+                BatchElementError.ErrorCode.kEntityAccessError, "Entities are read-only"
+            )
+            for idx in range(len(entityReferences)):
+                errorCallback(idx, result)
+            return
+
+        # Iterate over all the entity references, calling the correct
+        # error/success callbacks into the host.
+        # You should handle success/failure on an entity-by-entity
+        # basis, do not abort your entire operation because any single
+        # entity is malformed/can't be processed for any reason, use
+        # the error callback and continue.
+        for idx, ref in enumerate(entityReferences):
+            # It may be that one of the references you are provided is
+            # recognized for this manager, but has some syntax error or
+            # is otherwise incorrect for your specific resolve context.
+            # For example, an asset reference that specifies a version
+            # for an un-versioned entity could be considered malformed.
+            #
+            # N.B. It's not required to perform an explicit check here
+            # if this is naturally serviced during your backend lookup,
+            # the key is not to error the whole batch, but use the error
+            # callback for relevant references.
+            identifier_is_malformed = is_malformed_ref(ref)
+            if identifier_is_malformed:
+                error_result = BatchElementError(
+                    BatchElementError.ErrorCode.kMalformedEntityReference,
+                    "Entity identifier is malformed",
+                )
+                errorCallback(idx, error_result)
+            else:
+                # If our manager has the asset in question, we can
+                # let the host know which traits make up this specific
+                # entity.
+                if ref.toString() in managed_filesystem_locations:
+                    # This manager only manages location, and manages it
+                    # for every entity, so we can just build a success
+                    # result consisting of Managed + LocatableContent.
+                    # It is likely you will want to replace this with
+                    # a query into your backend system to inspect
+                    # specific entities.
+                    success_result = {ManagedTrait.kId, LocatableContentTrait.kId}
+                    if ref.toString() == "my_asset_manager:///anAsset2":
+                        # For the purpose of allowing the
+                        # apiComplianceSuite to test multiple distinct
+                        # entities, have the second entity also have a
+                        # version trait so there is a difference.
+                        success_result.add(VersionTrait.kId)
```

Comment:
[consider] I'm wary of reporting a trait through `entityTraits` that cannot be `resolve`d, it's setting a bad precedent in this tutorial/example.

Could we choose a trait that has no properties, instead?

Reply:
Aye, fair enough, although I will note the docs for entityTraits say explicitly that traits the manager is not capable of resolving is still a valid response to `kRead`.

Changed to `ConfigTrait`, (and also bumped the pinned version of Mediacreation so I could use it, seemed like a good idea)
https://github.com/OpenAssetIO/Template-OpenAssetIO-Manager-Python/pull/19/commits/50b094181eab70620a50b5c81c4e91d7ed4d9766

Reply:
:+1: 

Though I'm not sure its wise to say we're only adding the trait for `apiComplianceSuite`, might give the wrong impression.  We could say the second entity _is_ for an application configuration file (and as a nice side effect, it makes additional compliance suite tests work).

Also, if we really want to promote best practice through this, we should mix in the `Entity` trait to all of them.

Reply:
https://github.com/OpenAssetIO/Template-OpenAssetIO-Manager-Python/pull/19/commits/582b79f8c4f461e13aa5ba6e758783daf2539601

Reply:
Apologies, missed this before, but `Managed` is a policy trait, so shouldn't be returned from `entityTraits`.

Reply:
Nor should `Managed` be returned from `resolve` - it's purely a `managementPolicy` trait.

Reply:
I wonder if there's a way to api compliance that, if the traits themselves had understanding of what sort of traits they were.

Reply:
Removing ManagedTrait: 7df7e785f462881f8048850403e84694299b3b17